frontend/src/pages/library/LibraryPage.tsx

```diff
@@ -57,24 +63,45 @@ interface EntitiesTabProps {
 }
 
 function EntitiesTab({ render, renderAddButton }: EntitiesTabProps) {
-  const [entityTab, setEntityTab] = useState<EntityType>("project");
+  const { pathname } = useLocation();
+  const [entityTab, setEntityTab] = useState<Maybe<EntityType>>(undefined);
+
+  useEffect(() => {
+    setEntityTab(entitiesFlds.find(fld => pathname.endsWith(fld)));
+  }, [pathname]);
+
+  const renderRoute = (entity: EntityType) => (
+    <Route key={entity} path={`/library/${entity}`}>
+      {render(entity)}
+    </Route>
+  );
 
   return (
     <>
       <div className="entityTabs">
         {Object.entries(entities).map(([tpStr, title]) => (
-          <div
+          <Link
             key={tpStr}
+            to={`/library/${tpStr}`}
             className={cx("entityTabs__tab", entityTab === tpStr && "entityTabs__tab_active")}
             onClick={() => setEntityTab(tpStr as EntityType)}
           >
             {title}
-          </div>
+          </Link>
         ))}
 
-        {renderAddButton(entityTab)}
+        {entityTab && renderAddButton(entityTab)}
       </div>
-      {render(entityTab)}
+
+      <Switch>
+        {entitiesFlds.map(entity => renderRoute(entity))}
+
+        <Redirect exact from="/library" to="/library/projects" />
+
+        <Route path="*">
+          <NotFound />
+        </Route>
+      </Switch>
```

Comment:
Isn't it possible to put it to all other routers? Or it is known pattern?

Reply:
I've seen such patterns in react-router tutorial, where Switch is used not only as global router mechanism but some local one.

I don't know if it's known pattern or not, but why should we create this Library implementation details implementation? Considering this, I decided to put it here instead of putting directly in App. I